File: 153-find-minimum-in-rotated-sorted-array/153-find-minimum-in-rotated-sorted-array.java
class Solution {
    public int findMin(int[] arr) {
        int i=0,j=arr.length-1,res=Integer.MAX_VALUE;
        
        while(i<=j){
            int mid=(i+j)>>1;
            // System.out.println(i+" "+j+" "+mid);
            
            if(arr[j]<arr[mid])
                i=mid+1;
            else
                j=mid-1;
            res=Math.min(res,arr[mid]);
        }
        
        return res;
    }
}

File: 153-find-minimum-in-rotated-sorted-array/153-find-minimum-in-rotated-sorted-array.cpp
class Solution {
public:
    int findMin(vector<int>& arr) {
        int i=0,j=arr.size()-1,res=INT_MAX;
        
        while(i<=j){
            int mid=i+(j-i)/2;
            res=min(res,arr[mid]);
            
            if(arr[i]<=arr[mid]){
                if(arr[mid]>arr[j])
                    i=mid+1;
                else
                    j=mid-1;
            }
            else{
                j=mid-1;
            }
        }
        
        return res;
    }
};